### PR TITLE
Harden PXF's Tomcat configuration

### PR DESCRIPTION
**Pull request by frankgh**
- Remove default applications
- Remove Tomcat version information
- Set logs folder to 300

### CODE REVIEW

File: server/pxf-service/src/scripts/pxf-service [hunk 1]
```diff
@@ -118,6 +118,20 @@ function fail()
...
+function tomcatHardening()
+{
+    # Remove default/unwanted Applications
+    rm -rf "$tomcat_root"/webapps/ROOT
+    rm -rf "$tomcat_root"/webapps/docs
+    rm -rf "$tomcat_root"/webapps/examples
+    rm -rf "$tomcat_root"/webapps/host-manager
+    rm -rf "$tomcat_root"/webapps/manager
+}
+
```
Comment (denalex) [inline server/pxf-service/src/scripts/pxf-service]: this is good, but I think we should do it during Tomcat packaging into the build, it's a bit silly trying to remove something after we install it, and might be tricky for RPM install as well.

File: server/pxf-service/src/scripts/pxf-service [hunk 2]
```diff
@@ -214,7 +228,8 @@ function createLogsDir()
     if [[ ! -d $PXF_LOGDIR ]]; then
         echo "Creating PXF logs directory $PXF_LOGDIR ..."
         mkdir -p "$PXF_LOGDIR"
-        chmod 700 "$PXF_LOGDIR"
+        # only write/execute access
+        chmod 300 "$PXF_LOGDIR"
```
Comment (denalex) [inline server/pxf-service/src/scripts/pxf-service]: why so restrictive ? even owner will not be able to list files in this directory

File: server/pxf-service/src/templates/tomcat/conf/server.xml [hunk 3]
```diff
@@ -52,9 +52,13 @@ under the License.
...
+                       showReport="true"
+                       showServerInfo="false"/>
             </Host>
         </Engine>
-        <Connector acceptCount="100"
+        <Connector server="PXF Server"
+                   acceptCount="100"
                    connectionTimeout="20000"
                    executor="tomcatThreadPool"
                    maxKeepAliveRequests="15"
```
Comment (denalex) [inline server/pxf-service/src/templates/tomcat/conf/server.xml]: `web.xml` also needs to be updated, it contains JSP servlet, welcome pages, etc

File: server/pxf-service/src/templates/tomcat/conf/server.xml [hunk 4]
```diff
@@ -52,9 +52,13 @@ under the License.
                        pattern="%h %l %u %t &quot;%r&quot; %s %b"
```
Comment (denalex) [inline server/pxf-service/src/templates/tomcat/conf/server.xml]: we also do not use "in-memory user database" in lines 26-31

File: server/pxf-service/src/scripts/pxf-service (hunk 2 above)
Comment (frankgh) [reply server/pxf-service/src/scripts/pxf-service]: this is the recommended for log files, only write/execute access for the catalina user

Comment (frankgh) [reply server/pxf-service/src/scripts/pxf-service]: In my local environment, it has become annoying to tail the pxf logs. Even though 300 is recommended, maybe it's not practical?

Comment (oliverralbertini) [reply server/pxf-service/src/scripts/pxf-service]: How is it annoying, what's the workaround?

File: server/pxf-service/src/templates/tomcat/conf/server.xml [hunk 5]
```diff
@@ -52,9 +40,13 @@ under the License.
...
                        prefix="localhost_access_log."
                        suffix=".txt"/>
+                <Valve className="org.apache.catalina.valves.ErrorReportValve"
+                       showReport="true"
+                       showServerInfo="false"/>
             </Host>
         </Engine>
-        <Connector acceptCount="100"
+        <Connector server="PXF Server"
```
Comment (oliverralbertini) [inline server/pxf-service/src/templates/tomcat/conf/server.xml]: what does adding server field do for security?

File: server/pxf-service/src/scripts/pxf-service (hunk 2 above)
Comment (frankgh) [reply server/pxf-service/src/scripts/pxf-service]: sudo tail, but it's not practical, I agree. Maybe we should leave it at 700, but recommend 300 in the documentation?

Comment (frankgh) [reply server/pxf-service/src/scripts/pxf-service]: after the directory exists, `pxf init` will not do anything, so maybe it's okay to ship with 300 and let the users decide to change it to 700?

File: server/pxf-service/src/templates/tomcat/conf/server.xml (hunk 5 above)
Comment (frankgh) [reply server/pxf-service/src/templates/tomcat/conf/server.xml]: It displays `PXF Server` for the server value in the response headers
```
±  |harden-tomcat → origin {34} U:1 ?:20 ✗| → curl -i http://localhost:5888/pxf/v0
HTTP/1.1 500 Internal Server Error
Content-Type: text/plain
Transfer-Encoding: chunked
Date: Tue, 25 Feb 2020 13:15:23 GMT
Connection: close
Server: PXF Server
```

File: server/pxf-service/src/scripts/pxf-service (hunk 2 above)
Comment (denalex) [reply server/pxf-service/src/scripts/pxf-service]: let's not over-restrict things -- they should be usable out-of-the-box and users still can harden it further if required.

**Pull request merged**